scapy/arch/windows/__init__.py

```diff
@@ -83,55 +83,89 @@ def _windows_title(title=None):
         title = title or "Scapy v" + conf.version
         ctypes.windll.kernel32.SetConsoleTitleW(title)
 
-def _suppress_file_handles_inheritance(r=100):
+def _suppress_file_handles_inheritance(r=1000):
     """HACK: python 2.7 file descriptors.
 
     This magic hack fixes https://bugs.python.org/issue19575
     by suppressing the HANDLE_FLAG_INHERIT flag to a range of
     already opened file descriptors.
     """
-    # See https://github.com/secdev/scapy/issues/1136
+    # Fix https://bugs.python.org/issue19575
+    # and https://github.com/secdev/scapy/issues/1136
+    # Bug was fixed on python 3.4+
+
+    if sys.version_info[0:2] >= (3, 4):
+        return []
+
     import stat
     from ctypes import windll, wintypes
     from msvcrt import get_osfhandle
 
     HANDLE_FLAG_INHERIT = 0x00000001
 
+    handles = []
     for fd in range(r):
         try:
             s = os.fstat(fd)
         except OSError:
             continue
         if stat.S_ISREG(s.st_mode):
-            handle = wintypes.HANDLE(get_osfhandle(fd))
+            osf_handle = get_osfhandle(fd)
+            handle = wintypes.HANDLE(osf_handle)
+            flags  = wintypes.DWORD()
+            windll.kernel32.GetHandleInformation(handle, ctypes.byref(flags))
+            if flags.value & HANDLE_FLAG_INHERIT:
+                mask = wintypes.DWORD(HANDLE_FLAG_INHERIT)
+                flags = wintypes.DWORD(0)
+                windll.kernel32.SetHandleInformation(handle, mask, flags)
+                handles.append(osf_handle)
+
+    return handles
+
+def _restore_file_handles_inheritance(handles):
+    """HACK: python 2.7 file descriptors.
+
+    This magic hack fixes https://bugs.python.org/issue19575
+    by suppressing the HANDLE_FLAG_INHERIT flag to a range of
+    already opened file descriptors.
+    """
+    # See https://github.com/secdev/scapy/issues/1136
+    # Bug was fixed on python 3.4+
+
+    if sys.version_info[0:2] >= (3, 4):
+        return
+
+    from ctypes import windll, wintypes
+
+    HANDLE_FLAG_INHERIT = 0x00000001
+
+    for osf_handle in handles:
+        try:
+            handle = wintypes.HANDLE(osf_handle)
             mask = wintypes.DWORD(HANDLE_FLAG_INHERIT)
-            flags = wintypes.DWORD(0)
+            flags = wintypes.DWORD(HANDLE_FLAG_INHERIT)
```

Comment:
Please leave 1

Reply:
Better is HANDLE_FLAG_INHERITED.
Take a look at:
https://msdn.microsoft.com/pl-pl/library/windows/desktop/ms724935(v=vs.85).aspx

Reply:
ok